src/levanter/main/sft.py

```diff
@@ -0,0 +1,204 @@
+import logging
+import os
+from dataclasses import dataclass
+from enum import Enum
+from typing import List, Optional
+
+import jax.random as jrandom
+import transformers
+
+import haliax as hax
+from haliax import Axis
+from haliax.partitioning import round_axis_for_partitioning
+
+import levanter
+from levanter import callbacks
+from levanter.compat.hf_checkpoints import HFCheckpointConverter, HFCompatConfig, save_hf_checkpoint_callback
+from levanter.data import PermutationDataset
+from levanter.data.text import ChatSFTDatasetConfig, EpochDataset, mk_chat_sft_dataset, mk_supervised_dataset
+from levanter.main.train_lm import TrainLmConfig
+from levanter.models.lm_model import LmHeadModel, compute_next_token_loss
+from levanter.trainer import Trainer
+
+
+logger = logging.getLogger(__name__)
+
+# Define default special tokens
+DEFAULT_PAD_TOKEN = "[PAD]"
+DEFAULT_EOS_TOKEN = "</s>"
+DEFAULT_BOS_TOKEN = "<s>"
+DEFAULT_UNK_TOKEN = "<unk>"
+
+
+class DatasetType(str, Enum):
+    """Type of dataset to use"""
+
+    HUGGINGFACE = "huggingface"  # Use HF dataset
+    CHAT_JSONL = "chat_jsonl"  # Use JSONL files with chat format
+
+
+@dataclass
+class SFTConfig(TrainLmConfig):
```

Comment:
don't want to do this one?

Reply:
I end up needing a lot of shared content with TrainLMConfig since I want to explicitly declare all variables in the marin experiment so i think it's better to inherit

Reply:
generally speaking it's a code smell when you inherit and don't use the stuff in the super

Reply:
Got it, I just changed it to not inherit and copied the attributes I need